packages/grafana-data/src/types/datasource.ts

```diff
@@ -231,6 +231,15 @@ abstract class DataSourceApi<
    */
   abstract query(request: DataQueryRequest<TQuery>): Promise<DataQueryResponse> | Observable<DataQueryResponse>;
 
+  /**
+   * Returns observable to get histogram data. If undefined is returned, histogram is not available for
+   * the given DataQueryRequest.
+   *
+   * Subscription to the returned observable may be created along with the query request (if auto loading
+   * the histogram is enabled, or on demand when the user clicks on load logs volume button)
+   */
+  getLogsVolumeQuery?(request?: DataQueryRequest<TQuery>): Observable<DataQueryResponse> | undefined;
```

Comment:
Is this something that will implemented for most data sources?

Reply:
We will have it for Elastic and it could be used for any datasources returning logs. We're also considering more generic approach to return "related data" for a given query - https://github.com/grafana/grafana/pull/39042